File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/ui/search/base/SearchOnTheMapView.java
package com.example.ama.android2_lesson03.ui.search.base;

import android.net.Uri;

import com.example.ama.android2_lesson03.ui.base.PocketMapView;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.Marker;

public interface SearchOnTheMapView extends PocketMapView {
    void showOnGMapsApp(Uri uri);

    void showOnInnerMap(String markerTitle, String address, LatLng latLng);

    void requestPermission(String permission, int requestCode);

    void showMessage(String message);

    void moveMapCamera(LatLng latLng, float zoom, boolean cameraAnimation);

    void showDialog(String title, String message, Marker marker);
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/ui/search/base/SearchOnTheMapView.kt
package com.example.ama.android2_lesson03.ui.search.base

import android.net.Uri
import com.example.ama.android2_lesson03.ui.base.PocketMapView
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.Marker

interface SearchOnTheMapView : PocketMapView {
    fun showOnGMapsApp(uri: Uri)
    fun showOnInnerMap(markerTitle: String?, address: String, latLng: LatLng)
    fun showMessage(message: String)
    fun moveMapCamera(latLng: LatLng, zoom: Float, cameraAnimation: Boolean)
    fun requestPermission(permission: String, requestCode: Int)
    fun showEditDialog(dialogTitle: String, dialogMessage: String, marker: Marker)
}